### ISSUE TITLE
UWP Support

### ISSUE THREAD

**Issue by IncludeGuys** [guttir14/SoT-Hook#5]
After injecting cheat.dll with loader.exe and enabling global visuals in the menu there is still nothing displaying in the game. Btw can you tell me which version of windows are u using ?

**Comment by guttir14** [comment]
I use Windows 10 2004 currently. 

Can you provide more info about your issue? Did you enable other settings except 'Global Visual Enable' checkbox? 

![Image](https://i.imgur.com/CHLD5xt.png)

**Comment by IncludeGuys** [comment]
Yep I enabled every single checkbox in visuals menu and still nothing displaying in the game :/ Global visuals checkbox also enabled

**Comment by guttir14** [comment]
Do you use steam version of game?

**Comment by IncludeGuys** [comment]
Nope Windows Store version

**Comment by guttir14** [comment]
I didn't test cheat with Windows Store version. Are there 'Exception' lines in 'log.txt' file?

**Comment by IncludeGuys** [comment]
Where is that log.txt ? Cuz i dont have it in my folder with cheat

**Comment by guttir14** [comment]
It should be near Cheat.dll. It creates every time you inject the cheat.

**Comment by IncludeGuys** [comment]
Nope nothing were created

**Comment by IncludeGuys** [comment]
Btw I tried to load the dll with process hacker and after loading cheat.dll module I dont see it in module list is that normal ?

**Comment by guttir14** [comment]
It should be in module list if you inject with process hacker. Maybe there is something different in Microsoft Store package of game. Also, I can't understand why log.txt is not created for you.

**Comment by IncludeGuys** [comment]
My folder name with the cheat is called "new" so I don't think that is the problem. I think something is fucked with the injector cuz after using a loader or module load via process hacker I don't see any "cheat.dll" module.

**Comment by guttir14** [comment]
I have bought xbox pass and i'll try to figure out what problem

**Comment by IncludeGuys** [comment]
Ok thanks so much let me know if u fix the issue

**Comment by guttir14** [comment]
To add support for UWP version of Sea of Thieves I need to do a huge rework. I don't think I gonna do it shortly. 
For some reason UWP apps don't create usual WinAPI window, also they don't have access to entire filesystem. With this in head I need to rewrite the cheat.

**Comment by IncludeGuys** [comment]
Ok, thanks for letting me know what was the issue.

**Comment by IncludeGuys** [comment]
![Inject](https://user-images.githubusercontent.com/73388590/97095304-3b699480-164d-11eb-9ed3-499abdd3caf5.PNG)
Still nothing

**Comment by guttir14** [comment]
I didn't add UWP support to cheat yet. Only done UWP injector.

**Comment by guttir14** [comment]
I finally made game render my menu, now i need to come up with input system. 
![](https://cdn.discordapp.com/attachments/640480607402459136/770306252181209128/unknown.png)

**Comment by IncludeGuys** [comment]
Very nice man I'm waiting for update.

**Comment by guttir14** [comment]
As of today update cheat should be working in UWP version, but with some known bugs: 
- you can close/open menu only when you have started voyage.
- you can't type values with your keyboard
- bug with cursor (manual fix: minimize and maximize game to show or hide cursor)
- you can't save cheat settings

To compile cheat for UWP version set below 1 to 0 in cheat.cpp before compiling
```
#if 1
#define STEAM
#endif
```

**Comment by IncludeGuys** [comment]
Thanks for the update dude I'm still waiting for further improvements for me I can use menu even if I'm not in the voyage. After changing from fullscreen to borderless my games crashing. Btw which key activates aimbot?

**Comment by IncludeGuys** [comment]
Bug ~Aimbot works only if menu is active

**Comment by N0Name1991** [comment]
The menu doesn't show up at all for me it. It just makes my cursor disappear.

**Comment by guttir14** [comment]
My 14-day xbox game pass has already expired. doubt that I can continue research on the UWP version, I hope that some interested guy will do a PR with solving all these problems related to UWP or make a working fork.
